File: InsertionSort.java
// Insertion sort in java

// class definition
public class InsertionSort {
	// main program
	public static void main(String[] args) {
		int[] sampleList = {4,3,6,2,1,9,8,7,5,0};
		InsertionSort.performInsertionSort(sampleList);

		for (int e: sampleList) System.out.println(e);

	}

	/*
		function to sort a given list using the insertion method

		arrays in Java are mutable, no need to return the sorted list
	*/
	private static void performInsertionSort(int[] toSort) {
		// traverse through the contents of the list to be sorted
		for (int i=0; i<toSort.length; i++) {
			for (int j=i; j>=0; j--) {
				if (toSort[i] < toSort[j]) {
					int temp = toSort[i];
					toSort[i] = toSort[j];
					toSort[j] = temp;
					i = j;
				}
			}
		}
	}
}

File: InsertionSort.js
// Insertion Sort in Javascript

// define a function that performs Insertion sort
// no need to return anything since arrays in javascript are mutable
function performInsertionSort(toSort) {
    for (let i=1; i<toSort.length; i++) {
        let j=i;
        while (j>0 && toSort[j-1]>toSort[j]) {
            // perform swapping of adjacent elements
            temp = toSort[j];
            toSort[j] = toSort[j-1];
            toSort[j-1] = temp

            // update the value of j
            j--;
        }
    }
}


sampleArr = [3,2,5,6,7,8,9,1,0,4];
console.log(sampleArr);
performInsertionSort(sampleArr);
console.log(sampleArr);